### ISSUE TITLE
Separate out svg and gi3d to their own repositories, and Ki Trees.  Gi *exclusively* contains Widgets!

### ISSUE THREAD

**Issue by rcoreilly** [goki/gi#623]
A standalone SVG package, built on top of standalone GiRl #622, would be quite useful -- main other option is https://github.com/srwiley/oksvg but it is missing a fair bit of important stuff and isn't frequently updated.

It would make a lot more sense to have the SVG elements fully compartmentalized within their own standalone Ki tree, rather than being part of the overall widget scenegraph.  The parent SVGViewport just has an image that it renders to, and it doesn't need to bother about anything else except rendering straight to that image.  The gi.SVG as a widget just wraps around the SVG image, similar to bitmap.

SVG nodes do not directly process any events, so they really don't need to be in the overall tree.  The parent widget can handle any relevant events.  For example, grid exclusively processes mouse events at the level of the wrapping SVGView and uses sprites.

Likewise Gi3D can be its own separate repository, with 3D nodes in a completely separate tree.  The Embed2D node does get all events, but it wraps everything through the EmbedViewport, and other nodes only handle selection (manipulation is via additional separate manip nodes -- should put those in a separate tree too?).  In any case, Gi3D imports Gi and Gi contains the special support stuff to enable Gi3D -- mainly just separating out the impl so standard 2D users don't need to worry about it, and keeping the 3D tree separate enables the *Widget only* Gi.

Bottom line: Gi as a tree comprised exclusively of widgets!

Window is no longer a NodeBase -- just a standalone struct -- it doesn't have anything in common with a Widget Node -- and how about giving it a stack of Viewports which are also not nodes, but are rather things that contain widget trees and an associated image.  Each popup is a Viewport as it is now.  But having the widget trees all separated in their own Viewports makes so much more sense.    

Basically, went too far in having one huge monolithic tree -- more properly encapsulated trees with their own definite node type is more sensible.  No longer necessary to constantly check which type of node it is -- just treat everything as a widget, etc.

**Comment by rcoreilly** [comment]
Needed: name for SVG repo.  gsvg?  just svg?  probably just going for plain svg simplest and why not?
